chains/polygon/projects/siren/index.js

```diff
@@ -0,0 +1,145 @@
+/*==================================================
+  Modules
+  ==================================================*/
+const { request, gql } = require('graphql-request');
+const BigNumber = require('bignumber.js');
+const sdk = require('../../../../sdk');
+
+const POLYGON_GRAPH_URL = 'https://api.thegraph.com/subgraphs/name/sirenmarkets/protocol-v2-matic';
+
+/*==================================================
+  TVL
+  ==================================================*/
+
+async function getTokens(amms) {
+  let tokens = [];
+  for (let i = 0; i < amms['amms'].length; i++) {
+    let amm = amms['amms'][i];
+    tokens.push(amm['collateralToken']['id']);
+  }
+  return tokens;
+}
+
+async function getHolders(amms) {
+  const { seriesVaults } = await request(POLYGON_GRAPH_URL, GET_SERIES_VAULT_ID);
+
+  const holders = [];
+
+  for (let i = 0; i < amms['amms'].length; i++) {
+    let amm = amms['amms'][i];
+    holders.push(amm['id']);
+  }
+  holders.push(seriesVaults[0]['id']);
+  return holders;
+}
+
+async function calculatePolygonTVL(timestamp, block, polygon_amms) {
+  const { seriesVaults } = await request(POLYGON_GRAPH_URL, GET_SERIES_VAULT_ID);
+
+  const polygon_result = {
+    'polygon:0x2791bca1f2de4661ed88a30c99a7a9449aa84174': '0'
+  };
+
+  for (let i = 0; i < polygon_amms['amms'].length; i++) {
+    const amm = polygon_amms['amms'][i];
+    const collateralToken = amm['collateralToken']['id'];
+
+    // Get collateral in AMM
+    const response = await sdk.api.abi.call({
+      target: collateralToken,
+      params: amm['id'],
+      abi: 'erc20:balanceOf',
+      block: block
+    });
+
+    if (!polygon_result[collateralToken]) {
+      polygon_result[collateralToken] = '0';
+    }
+    polygon_result[collateralToken] = BigNumber(polygon_result[collateralToken]).plus(response.output).toFixed();
+
+    // Get collateral in Series
+    const response2 = await sdk.api.abi.call({
```

Comment:
have you verified this works?

Polygon sdk calls need param `chain: 'polygon'`

Reply:
The end result of the polygon is just a token holder map, this is code they used todo, not used in the adapter it self